Editor/Window/Components/StatusIndicator.cs

```diff
@@ -22,15 +23,33 @@ public StatusIndicator()
         {
             var uxml = Resources.Load<VisualTreeAsset>("EditorWindow/Components/StatusIndicator");
             uxml.CloneTree(this);
+            AddToClassList("status-indicator");
             AddToClassList("separator");
             AddToClassList("separator--horizontal");
         }
 
+        private void AddSpin()
+        {
+            schedule.Execute(() => { _imageContainer.AddToClassList("status-indicator--spin"); }).StartingIn(10);
+        }
+
         public void Set(State state, string text)
         {
             Reset();
             _label.text = text;
             AddToClassList(_stateClassNames[state]);
+            if (state == State.InProgress)
+            {
+                _imageContainer.RegisterCallback<TransitionEndEvent>(evt =>
+                {
+                    _imageContainer.RemoveFromClassList("status-indicator--spin");
```

Comment:
Shouldn't we also stop the spinning animation as soon as we transition to a state that isn't in progress? Otherwise, they'll be a weird moment where the success or failed icons are spinning

Reply:
Fixed!